content/chapters/software-stack/lecture/slides/types-of-software.md

```diff
@@ -0,0 +1,62 @@
+---
+
+## Types of Software
+
+----
+
+### Usable vs Reusable
+
+* Software provides features by extending hardware
+* Features are used by:
+  * users (as interactive applications / processes)
+  * system (as non-interactive services / processes)
+  * other applications (as reusable libraries)
+* Software components
+  * Applications: usable
+  * Libraries: reusable
+
+----
+
+### Applications
+
+![Applications](media/applications.svg)
+
+----
+
+### Libraries
+
+![Libraries](media/libraries.svg)
+
+----
+
+### Application Software and System Software
+
+![Application and System Software](media/application-system-software.png)
+<!-- https://www.webopedia.com/definitions/systems-software/ -->
+
+----
+
+### Libraries and Frameworks
+
+![Libraries and Frameworks](media/libraries-frameworks.png)
```

Comment:
This image says "Your Code", but that's unfair. "Your Code" could be in any of the components. Redo this image yourself and say "Application Code" instead.